crates/ethereum/engine/src/orchestrator.rs

```diff
@@ -0,0 +1,224 @@
+use futures::StreamExt;
+use pin_project::pin_project;
+use reth_beacon_consensus::{BeaconEngineMessage, EthBeaconConsensus};
+use reth_chainspec::ChainSpec;
+use reth_db_api::database::Database;
+use reth_engine_primitives::EngineTypes;
+use reth_engine_tree::{
+    backfill::PipelineSync,
+    chain::ChainOrchestrator,
+    download::BasicBlockDownloader,
+    engine::{EngineApiEvent, EngineApiRequestHandler, EngineHandler, FromEngine},
+};
+use reth_ethereum_engine_primitives::EthEngineTypes;
+use reth_network_p2p::{bodies::client::BodiesClient, headers::client::HeadersClient};
+use reth_stages_api::Pipeline;
+use reth_tasks::TaskSpawner;
+use std::{
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::UnboundedReceiverStream;
+
+/// Alias for Ethereum chain orchestrator.
+type EthChainOrchestratorType<DB, Client> = ChainOrchestrator<
+    EngineHandler<
+        EngineApiRequestHandler<EthEngineTypes>,
+        UnboundedReceiverStream<BeaconEngineMessage<EthEngineTypes>>,
+        BasicBlockDownloader<Client>,
+    >,
+    PipelineSync<DB>,
+>;
+
+/// Builder for `EthChainOrchestrator`.
+#[allow(missing_debug_implementations, dead_code)]
+pub struct EthChainOrchestratorBuilder<DB, Client>
+where
+    DB: Database + 'static,
+    Client: HeadersClient + BodiesClient + Clone + Unpin + 'static,
+{
```

Comment:
I think this builder type will be slightly inconvenient to use because of the generics.

by the time we're creating this:

https://github.com/paradigmxyz/reth/blob/405b73045549611c3381020cef9302b4ee3f483c/crates/node/builder/src/launch/mod.rs#L250-L263

we should have all of the inputs available already, so I'm fine with a large `fn new` for now